packages/admin/cms-admin/src/common/MasterMenu.tsx

```diff
@@ -17,64 +20,97 @@ type MasterMenuItemRoute = Omit<MenuItemRouterLinkProps, "to"> & {
     requiredPermission?: string;
     route?: RouteProps;
     to?: string;
-    submenu?: MasterMenuItem[];
+    submenu?: MasterMenuElement[];
 };
 
 type MasterMenuItemAnchor = MenuItemAnchorLinkProps & {
     requiredPermission?: string;
 };
 
+type MasterMenuItemGroup = MenuItemGroupProps & {
+    requiredPermission?: string;
+    groupItems: MasterMenuItem[];
+};
+
 export type MasterMenuItem = MasterMenuItemRoute | MasterMenuItemAnchor;
 
-export type MasterMenuData = MasterMenuItem[];
+export type MasterMenuElement = MasterMenuItemRoute | MasterMenuItemAnchor | MasterMenuItemGroup;
```

Comment:
My thought behind it was that MenuItemGroups are technically no MenuItems. Thats why I wanted to separate it.

Reply:
But it is an item of `MasterMenuData`, so I'd say it could also be part of `MasterMenuItem`, which is not to confused with `MenuItem` 😬 

Having both `MasterMenuItem` and `MasterMenuElement` is confusing, don't you think?

Reply:
Yeah a bit thats true. Changed it here https://github.com/vivid-planet/comet/pull/2080/commits/47cbf507e55c04cdd17495d8f74b9d3dcd41065a